Update rLogin docs (#744)
rsksmart/devportal: rif/rlogin/get-started.md

title: rLogin - login tool for RSK
---

Integrate rLogin into your app and allow your users to choose their favorite wallets to login. With a single tool you will get connected to their wallet using an API compatible with Metamask, continue developing as you did.

<iframe src="https://codesandbox.io/embed/rlogin-9mmoc?fontsize=14&hidenavigation=1&theme=dark"
     style="width:100%; height:500px; border:0; border-radius: 4px; overflow:hidden;"
     title="rlogin"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>

<div class="row">
  <div class="col">
    <h3>Wallet support:</h3>
    <ul>
      <li>Browser wallets: Metamask, Nifty, Liquality</li>
      <li>Mobile wallets via Wallet Connect</li>
      <li>Custodial wallets: Portis, Torus (beta)</li>
      <li>Hardware wallets: Ledger, Trezor, D'Cent</li>
    </ul>
    <h3>EIP-1193 support</h3>
  </div>
  <div class="col">
    <h3>Network support:</h3>
    <ul>
      <li>RSK Mainnet, RSK Testnet</li>

>       <li>Etheruem, Ropsten, Kovan, Rinkeby, Gorely</li>

Reviewer: ```suggestion
      <li>Ethereum, Ropsten, Kovan, Rinkeby, Goerli</li>
```

  import { ledgerProviderOptions } from '@rsksmart/rlogin-ledger-provider'
  import { dcentProviderOptions } from '@rsksmart/rlogin-dcent-provider'

  const rpcUrls = {
    30: 'https://public-node.rsk.co',
    31: 'https://public-node.testnet.rsk.co',
  }

  const supportedChains = Object.keys(rpcUrls).map(Number)

  export const rLogin = new RLogin({
    providerOptions: {
      walletconnect: {
        package: WalletConnectProvider,
        options: {
          rpc: rpcUrls
        }
      },
      portis: {
        package: Portis,
        options: {
          id: "a1c8672b-7b1c-476b-b3d0-41c27d575920",
          network: {
            nodeUrl: 'https://public-node.testnet.rsk.co',
            chainId: 31,
          }
        }
      },
      torus: {
          package: Torus,
      },
      'custom-ledger': ledgerProviderOptions,
      'custom-dcent': dcentProviderOptions,
      'custom-trezor': {
        ...trezorProviderOptions,
        options: {
          manifestEmail: 'info@iovlabs.org',
          manifestAppUrl: 'https://basic-sample.rlogin.identity.rifos.org/',
        }
      }
    },
    rpcUrls,
    supportedChains
  })
  ```

>   > We usually put this all together in a single file called `rlogin.ts` and export a single instance of `RLogin`. This ensures a single DOM element is creted.

Reviewer: ```suggestion
  > Tip: Put this all together in a single file named `rlogin.ts` and export a single instance of `RLogin`. This ensures that only a single DOM element is created.
```

- 3) Connect!
  ```typescript
  import { providers } from 'ethers'

  const login = () => rLogin.connect()
      .then(({ provider, disconnect }) => {
          const provider = new providers.Web3Provider(provider)
          provider.getSigner(0).getAddress().then(console.log)
      })
  ```

>   You can use `provider` with your client of preference: [`Web3.js`](https://github.com/ethereum/web3.js/), [`ethjs`](https://github.com/ethjs/ethjs), [`ethers.js`](https://github.com/ethers-io/ethers.js/) or other.

Reviewer 2: ```suggestion
  You can use `provider` with your preferred client: [`Web3.js`](https://github.com/ethereum/web3.js/), [`ethjs`](https://github.com/ethjs/ethjs), [`ethers.js`](https://github.com/ethers-io/ethers.js/) or other.
```

<div class="container the-stack">
  <div class="row rif_blue_text">
    <div class="col">
      <div class="rns-index-box">
        <a href="../features">Features</a>
        <br />
        <br />
        <p>i18n, theming, dark/light, listeners, triggers</p>
      </div>
    </div>
    <div class="col">
      <div class="rns-index-box">
        <a href="../samples">rLogin sample apps</a>
        <br />
        <br />
        <p>Sample apps! Find all the code you need</p>
      </div>
    </div>
  </div>
  <div class="row rif_blue_text">
    <div class="col">
      <div class="rns-index-box">
        <a href="../authentication">Integrated backend authentication</a>
        <br />
        <br />
        <p>Integrate the authentication model based on the user's digital signature capabilities</p>
      </div>
    </div>
    <div class="col">
      <div class="rns-index-box">
        <a href="../migrating">Migrating</a>
        <br />
        <br />
        <p>From <code>web3modal</code> or <code>web3react</code></p>
      </div>
    </div>
  </div>
</div>

## EIP-1193

> rLogin supports [EIP-1193](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md). This means rLogin `provider` is comaptible with the API discribed in the standard, thus polymorphic with Metamask API. This also allows you to make a single implementation and integrate all the supported wallets instantly, and let you work with your favourite web3 client.

Reviewer: also consider that you talk about improved user experience in the next section. in this section you could claim that it provides an improved developer experience.

```suggestion
rLogin supports [EIP-1193](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md). This means rLogin `provider` is compatible with the API described in the standard, thus polymorphic with Metamask API. This also allows you to make a single implementation and integrate all the supported wallets instantly, and let you work with your favourite web3 client.
```

Author: See https://github.com/rsksmart/devportal/pull/744/files/2c12af2e974f622314c534f148fcad65c0bb0cf5#r723316011

## The UX

The UX is the most important part of this project. This is the main point to consider integrating rLogin

### Wallet information

Before logging in to the app the user will be able to validate the account and network they are connecting to.

<p align="middle">
  <img src="https://i.imgur.com/OZjZZPb.png" alt="wallt info" height="400" />
</p>

### Tutorials and helpers for hardware wallets

> Hardware wallets can be a headache... the rLogin includes some small tutorials that will help a lot the user to get connected to the different networks with the most secure devices.

Reviewer: ```suggestion
Hardware wallets can be quite hard to use. rLogin includes a few short tutorials that will help users to overcome these difficulties, and get connected to the different networks with the most secure devices.
```

<p align="middle">
  <img src="https://i.imgur.com/dqGd4qO.png" alt="wallt info" height="350" />
</p>

### Choose network

Some wallets need the user to pick their wallet network conneciton from the UI. We added a selector for the networks that are listed in `rpcUrls` This step will be prompted for hardware wallets or Torus.

<p align="middle">
  <img src="https://i.imgur.com/KkJxOn0.png" alt="wallt info" height="200" />
</p>

### Supported networks

If the user can choose the network in their wallet rLogin will ask the user to change the network before `rLogin.connect()` happens. This will ensure you the user is connected to the correct network when landing.

> It will also show up after logging in if the user changes the network and it is not supported.

Reviewer: ```suggestion
It will also show up after logging in, if the user changes the network and it is not supported.
```
